jina/clients/base.py

```diff
@@ -151,6 +154,8 @@ async def _get_results(self, input_fn: Callable,
                                      'please double check your input iterator') from rpc_ex
             else:
                 raise BadClient(msg) from rpc_ex
+        if self.args.return_results:
+            return result
```

Comment:
would it make sense to return a generator to make it more general?

Reply:
tried. too complicated., async generator is depreciated in python asyncio

Reply:
also, if async generator is desired, why not directly use `on_done`?

Reply:
see design decision in PR body

Reply:
that is true. Just a comment that felt more natural, but no big deal